ml-proto/test/conversions.wasm

```diff
@@ -89,14 +146,113 @@
 (assert_eq (invoke "$i64_trunc_u_f32" (f32.const 1.5)) (i64.const 1))
 (assert_eq (invoke "$i64_trunc_u_f32" (f32.const 4294967296)) (i64.const 4294967296))
 
-(;
-  todo:
-    fxx.convert_s/ixx
-    fxx.convert_u/ixx
-    f64.promote/f32
-    f32.demote/f64
-    f32.reinterpret/i32
-    f64.reinterpret/i64
-    i32.reinterpret/f32
-    i64.reinterpret/f64
-;)
+;; f32.convert_s/i32
+(assert_eq (invoke "$f32_convert_s_i32" (i32.const 1)) (f32.const 1.0))
+(assert_eq (invoke "$f32_convert_s_i32" (i32.const -1)) (f32.const -1.0))
+(assert_eq (invoke "$f32_convert_s_i32" (i32.const 0)) (f32.const 0.0))
+(assert_eq (invoke "$f32_convert_s_i32" (i32.const 2147483647)) (f32.const 2147483647))
+(assert_eq (invoke "$f32_convert_s_i32" (i32.const -2147483648)) (f32.const -2147483648))
+(assert_eq (invoke "$f32_convert_s_i32" (i32.const 1234567890)) (f32.const 1234567890))
+
+;; f32.convert_s/i64
+(assert_eq (invoke "$f32_convert_s_i64" (i64.const 1)) (f32.const 1.0))
+(assert_eq (invoke "$f32_convert_s_i64" (i64.const -1)) (f32.const -1.0))
+(assert_eq (invoke "$f32_convert_s_i64" (i64.const 0)) (f32.const 0.0))
+(assert_eq (invoke "$f32_convert_s_i64" (i64.const 9223372036854775807)) (f32.const 9223372036854775807))
+(assert_eq (invoke "$f32_convert_s_i64" (i64.const -9223372036854775808)) (f32.const -9223372036854775808))
+(assert_eq (invoke "$f32_convert_s_i64" (i64.const 314159265358979)) (f32.const 314159265358979)) ;; PI
+
+;; f64.convert_s/i32
+(assert_eq (invoke "$f64_convert_s_i32" (i32.const 1)) (f64.const 1.0))
+(assert_eq (invoke "$f64_convert_s_i32" (i32.const -1)) (f64.const -1.0))
+(assert_eq (invoke "$f64_convert_s_i32" (i32.const 0)) (f64.const 0.0))
+(assert_eq (invoke "$f64_convert_s_i32" (i32.const 2147483647)) (f64.const 2147483647))
+(assert_eq (invoke "$f64_convert_s_i32" (i32.const -2147483648)) (f64.const -2147483648))
+(assert_eq (invoke "$f64_convert_s_i32" (i32.const 987654321)) (f64.const 987654321))
+
+;; f64.convert_s/i64
+(assert_eq (invoke "$f64_convert_s_i64" (i64.const 1)) (f64.const 1.0))
+(assert_eq (invoke "$f64_convert_s_i64" (i64.const -1)) (f64.const -1.0))
+(assert_eq (invoke "$f64_convert_s_i64" (i64.const 0)) (f64.const 0.0))
+(assert_eq (invoke "$f64_convert_s_i64" (i64.const 9223372036854775807)) (f64.const 9223372036854775807))
+(assert_eq (invoke "$f64_convert_s_i64" (i64.const -9223372036854775808)) (f64.const -9223372036854775808))
+(assert_eq (invoke "$f64_convert_s_i64" (i64.const 4669201609102990)) (f64.const 4669201609102990)) ;; Feigenbaum
+
+;; f32.convert_u/i32
+(assert_eq (invoke "$f32_convert_u_i32" (i32.const 1)) (f32.const 1.0))
+(assert_eq (invoke "$f32_convert_u_i32" (i32.const 0)) (f32.const 0.0))
+(assert_eq (invoke "$f32_convert_u_i32" (i32.const 2147483647)) (f32.const 2147483647))
+(assert_eq (invoke "$f32_convert_u_i32" (i32.const -2147483648)) (f32.const 2147483648))
+(assert_eq (invoke "$f32_convert_u_i32" (i32.const 305419896)) (f32.const 305419896)) ;; 0x12345678
```

Comment:
The wasm parser does recognize hexadecimal constants now, so you can write the constant directly now instead of writing it in decimal and having a comment :-).